src/bot.jl

```diff
@@ -8,24 +8,36 @@ const UStrings = Union{AbstractString,Regex,AbstractChar}
 Config object that holds the options and configuration for the SnoopCompile bot. This object is fed to the `@snoopi_bot`.
 
 # Arguments:
-- `packageName::String`
-- `subst::Vector{Pair{UStrings, UStrings}}` : to replace a packages precompile setences with another's package like `["ImageTest" => "Images"]`
-- `blacklist::Vector{UStrings}` : to remove some precompile sentences
+- `package_name::String`
 
-`const UStrings ==  Union{AbstractString,Regex,AbstractChar}` # every string like type that `replace()` has a method for.
+## optional:
+
+- `subst` : A vector of pairs of Strings (or RegExp) to replace a packages precompile setences with another's package like `["ImageTest" => "Images"]`.
+
+- `blacklist` : A vector of of Strings (or RegExp) to remove some precompile sentences
+
+- `os`: A vector of of Strings (or RegExp) to give the list of os that you want to generate precompile signatures for. Each element will call a `Sys.is\$eachos()` function.
+
+
+# Example
+```julia
+BotConfig("MatLang", blacklist = ["badfunction"], os = ["linux", "windows"])
+```
 """
 struct BotConfig
-    packageName::String
+    package_name::String
     subst::Vector{Pair{T1, T2}} where {T1<:UStrings, T2 <: UStrings}
```

Comment:
```suggestion
    subst::Vector{Pair{UStrings, UStrings}}
```
The compiler won't get any benefit from your parametric form since the parameters don't propagate to `BotConfig` itself. By making it broader you would allow one to push new items onto the list without worrying about whether the original type was incompatibility (e.g., if you create it with a `String=>String` type and then push on a Regex-based pair).